### PR TITLE
vmm: Restore tty to canonical mode when SIGTERM or SIGINT received

### PR DESCRIPTION
**Pull request by crab2313**
The tty mode remains raw mode when cloud-hypervisor is terminted by SIGTERM or SIGINT. The terminal is unusable due to echoing is disabled which is really annoying.

### CODE REVIEW

File: vmm/src/vm.rs [hunk 1]
```diff
@@ -675,12 +675,21 @@ impl Vm {
...
             }
+            if signal == SIGTERM || signal == SIGINT {
+                if on_tty {
+                    io::stdin()
+                        .lock()
+                        .set_canon_mode()
+                        .expect("failed to restore terminal mode");
+                }
+                std::process::exit((signal != SIGTERM) as i32);
+            }
```
Comment (sboeuf) [inline vmm/src/vm.rs]: Could you please use a `match` for both `SIGWINCH` and `SIGTERM | SIGINT` as this is more idiomatic for Rust.

Comment (crab2313) [reply vmm/src/vm.rs]: ok

**Comment by crab2313** [comment]
@sboeuf I can reproduce the issue simply by running `killall cloud-hypervisor` in another terminal. My environment is the latest archlinux by the way. 

```shell
#!/bin/bash

./cloud-hypervisor/target/release/cloud-hypervisor -vv\
        --api-socket /tmp/ch-socket \
        --kernel ./vmlinux.bin \
        --disk path=clear-29160-kvm.img \
        --cmdline "console=ttyS0 reboot=k panic=1 nomodules i8042.noaux i8042.nomux i8042.nopnp i8042.dumbkbd root=/dev/vda3" \
        --cpus 4 \
        --memory size=512M \
        --rng
```

**Comment by sboeuf** [comment]
@crab2313 interesting as I cannot reproduce (even using your command line) on Ubuntu.
Let's merge anyway as I think the PR looks good and both @rbradford and @sameo approved :)

**Pull request merged**